### PR TITLE
rust: fix lint warnings about if same then else -v7

### PR DESCRIPTION
**Pull request by Rai0Catodic0**
Ticket: [4609](https://redmine.openinfosecfoundation.org/issues/4609)

Make sure these boxes are signed before submitting your Pull Request -- thank you.

[ :white_check_mark: ]  I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
    [  :white_check_mark: ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
 - [ ]  I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: #8021 

Describe changes:
- Refactor range.rs to a more readable version

### CODE REVIEW

File: rust/src/http2/range.rs [hunk 1]
```diff
@@ -131,13 +131,16 @@ pub fn http2_range_open(
     tx: &mut HTTP2Transaction, v: &HTTPContentRange, flow: *const Flow,
     cfg: &'static SuricataFileContext, dir: Direction, data: &[u8],
 ) {
-    if v.end <= 0 || v.size <= 0 {
-        // skipped for incomplete range information
+    if v.end <= 0 || v.size <= 0{
```
Comment (inashivb) [inline rust/src/http2/range.rs:134]: nit: there should be a space between the condition and the brace

File: rust/src/http2/range.rs [hunk 2]
```diff
@@ -131,13 +131,16 @@ pub fn http2_range_open(
...
+        // skipped for incomplete range information
```
Comment (inashivb) [inline rust/src/http2/range.rs:135]: nit: unnecessary space added at the end

File: rust/src/http2/range.rs [hunk 3]
```diff
@@ -131,13 +131,16 @@ pub fn http2_range_open(
...
-        // skipped for incomplete range information
+    if v.end <= 0 || v.size <= 0{
+        // skipped for incomplete range information 
         return;
-    } else if v.end == v.size - 1 && v.start == 0 {
+    }
+    if v.end == v.size - 1 && v.start == 0 {
         // whole file in one range
         return;
+
```
Comment (inashivb) [inline rust/src/http2/range.rs:141]: nit: unnecessary newline

File: rust/src/http2/range.rs [hunk 4]
```diff
@@ -131,13 +131,16 @@ pub fn http2_range_open(
...
     }
+
```
Comment (inashivb) [inline rust/src/http2/range.rs:143]: nit: unnecessary newline

**Review by inashivb — CHANGES_REQUESTED** [review]
Hi @Rai0Catodic0 !
We're almost there. Just some formatting changes need to be made. Good job incorporating the feedbacks! :)

**Comment by inashivb** [comment]
@Rai0Catodic0 could you please make sure to close your previous revs of a PR when you create a new one? That helps keep only the latest PRs in view. :octocat:

**Comment by Rai0Catodic0** [comment]
Closing this in favour of newer PR #8035

**Pull request closed without merging**